fix(input)!: focus states by dropping support for UC & Opera Mini

pretzelhammer opened the pull request:
<!--
  🤖 This repo uses Conventional Commits (conventionalcommits.org) to automate
  release notes and versioning. Title your PR using the following template:

  <type>(<scope>): <subject>

  Scope is optional. Indicate a breaking change by adding ! after the type/scope.

  Version influencing types:
  - fix: user-facing bug fix (patch version bump)
  - feat: user-facing feature (minor version bump)

  Other types:
  - docs: changes to the documentation
  - build: changes that affect the build system or external dependencies
  - test: adding missing tests, refactoring tests; no production code change
  - refactor: refactoring production code, eg. renaming a variable
  - style: formatting, missing semi colons, etc; no production code change
  - chore: updating grunt tasks etc; no production code change
  - revert: reverts a previous commit
  - perf: changes that improve performance
  - ci: changes to CI configuration files and scripts (eg. GitHub Actions)

  👍 Do examples:
  - feat(button): primary variant
  - fix(action-bar): inherit event-listeners

  👎 Don't examples:
  - feat(button): [ABC-123] primary variant

  Read CONTRIBUTING.md for more info.
-->

## Describe the problem this PR addresses
<!--
  🤐 If you are a Square employee, be mindful of any internal information
  you share in this public repository.
-->
https://github.com/square/maker/issues/226#issuecomment-1071061425

## Describe the changes in this PR
<!--
  📸 Inline screenshots to better communicate the changes
-->
Drops support for UC Browser 12.12 and Opera Mini browsers so that our `:focus-within` CSS styles within the MInput component work again.

## Other information
<!--
  🙆‍♂️ Provide further context that will help those out-of-the-loop
  to quickly understand the changes.
-->
This has been broken since v4.1.0 when [this line](https://github.com/square/maker/compare/v4.0.1...v4.1.0#diff-43302d1852d8010ee012aab4dda2d01abb664e11f73617fe625f2ba112a958edR3) was released 🤦

landondurnan reviewed (approved):
Browserlist appears to work differently than I had expected. I didn't think it would remove unsupported code, rather I thought it'd simply warn of unsupported code.

It seems a shame to delist browsers because of a single CSS pseudo-element. We could target based on that pseudo-element, but then we're setting ourselves up to be far more granular about every new feature we want to use. 

It might be worth starting a discussion about how we could try to mitigate these types of issues in the future. I'll start one soon.

landondurnan commented:
@pretzelhammer Have we considered switching the `focus-within` to `focus`?

pretzelhammer commented:
@landondurnan we could maybe switch `focus-within` and use a combination of JS and `focus` to get a similar-ish effect but I'd rather just drop support for browsers that we never really meant to support in the first place.

According to our pubsite usage stats that Lokesh compiled a while back 0% of traffic comes from UC Browser and only 0.5% of traffic comes from Opera (and that's for _all_ Opera versions, Mini (for mobile devices) _and_ regular (for Desktop)). For reference, we dropped support for IE when it was 3% of our pubsite traffic.

Also, I was part of the discussion when we added the `> 0.5%` figure to our browserlist and the main reason it was added was to support older Chrome versions. We get significantly more pubsite traffic from older Chromes (more than 2 major versions old) than we get from UC and Opera combined. Older Chrome versions account for something like ~10% of traffic, and like I said before, UC and Opera combined are 0.5% (with 0% from UC and Opera usage going down every year).
